controlplane/kubeadm/api/v1beta1/v1beta2_condition_consts.go

```diff
@@ -18,41 +18,101 @@ package v1beta1
 
 import clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
 
-// Conditions that will be used for the KubeadmControlPlane object in v1Beta2 API version.
+// KubeadmControlPlane's Available condition and corresponding reasons that will be used in v1Beta2 API version.
 const (
 	// KubeadmControlPlaneAvailableV1Beta2Condition True if the control plane can be reached, EtcdClusterAvailable is true,
 	// and CertificatesAvailable is true.
 	KubeadmControlPlaneAvailableV1Beta2Condition = clusterv1.AvailableV1Beta2Condition
+)
 
+// KubeadmControlPlane's CertificatesAvailable condition and corresponding reasons that will be used in v1Beta2 API version.
+const (
 	// KubeadmControlPlaneCertificatesAvailableV1Beta2Condition True if all the cluster certificates exist.
 	KubeadmControlPlaneCertificatesAvailableV1Beta2Condition = "CertificatesAvailable"
 
+	// KubeadmControlPlaneCertificatesInternalErrorV1Beta2Reason surfaces unexpected failures when reconciling cluster certificates.
+	KubeadmControlPlaneCertificatesInternalErrorV1Beta2Reason = clusterv1.InternalErrorV1Beta2Reason
+
+	// KubeadmControlPlaneCertificatesAvailableV1Beta2Reason surfaces when cluster certificates are available,
+	// no matter if those certificates have been provided by the user or generated by  KubeadmControlPlane itself.
+	// Cluster certificates include: certificate authorities for ca, sa, front-proxy, etcd, and if external etcd is used,
+	// also the apiserver-etcd-client client certificate.
+	KubeadmControlPlaneCertificatesAvailableV1Beta2Reason = clusterv1.MachineAvailableV1Beta2Reason
```

Comment:
Should we really alias the KCP Certificate Available reason to the Machine available reason?